src/tests/TestCryptographicHash.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2005-2017 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include <QtCore>
+#include <QtTest>
+#include <QLatin1String>
+
+#include "CryptographicHash.h"
+
+class TestCryptographicHash : public QObject {
+		Q_OBJECT
+	private slots:
+		void sha1_data();
+		void sha1();
+
+		void sha256_data();
+		void sha256();
+
+		void staticHashVsObject();
+		void multipleResultCalls();
+		void addDataAfterResult();
+};
+
+/// normalizedHash removes all whitespace from the hex-encoded hash string.
+static QString normalizedHash(QString str) {
+	str.replace(QLatin1String(" "), QLatin1String(""));
+	str.replace(QLatin1String("\t"), QLatin1String(""));
+	str.replace(QLatin1String("\r"), QLatin1String(""));
+	str.replace(QLatin1String("\n"), QLatin1String(""));
+	return str;
+}
+
+static QByteArray normalizedHash(QByteArray buf) {
```

Comment:
Function names should be verbs, as they *do* stuff. So I would have expected this function to be called `normalizeHash`.